helix-term/src/commands.rs

```diff
@@ -2077,18 +2077,27 @@ pub mod cmd {
         Ok(())
     }
 
-    fn buffer_close_impl(
+    fn buffer_close_by_ids_impl(
         editor: &mut Editor,
-        args: &[Cow<str>],
+        doc_ids: impl IntoIterator<Item = DocumentId>,
```

Comment:
I was following the advice of the [API Guidelines](https://rust-lang.github.io/api-guidelines/flexibility.html#functions-minimize-assumptions-about-parameters-by-using-generics-c-generic), which I thought makes it so you can pass any sort of collection. Figured if we wanted to get rid of vec in some instance we could do it while retaining it where it's needed.

Reply:
In particular I think the `buffer_gather_paths_impl` needs to build a vec since we don't know how many of the args are valid. On the other hand, `buffer_gather_others_impl` might be able to get away with returning the chained iterator without collecting to a vec first, although I couldn't quite figure out how to do that ..

Reply:
This makes sense for a library API but in this case we can always change the method bounds in the future.

Reply:
Fixed in [9a94e1d](https://github.com/helix-editor/helix/pull/1677/commits/9a94e1d0880d044e2000932726d3147411b8f72d)